File: solutions/r/rna-transcription/1/rna-transcription.R
to_rna <- function(dna) {
  rna <- unlist(strsplit(dna,""))
  for(i in 1:length(rna)) {
    if(rna[i]=="G") {
      rna[i] <- "C"
    }
    else if(rna[i]=="C") {
      rna[i] <- "G"
    }
    else if(rna[i]=="A") {
      rna[i] <- "U"
    }
    else if(rna[i]=="T") {
      rna[i] <- "A"
    }
    else {
      stop("Invalid nucleotide.")
    }
  }
  rna <- paste0(rna,collapse="")
  return(rna)
}

File: solutions/julia/rna-transcription/1/rna-transcription.jl
function choose(nucleotide)
    if nucleotide == 'A'
        return 'U'
    elseif nucleotide == 'C'
        return 'G'
    elseif nucleotide == 'G'
        return 'C'
    elseif nucleotide == 'T'
        return 'A'
    end
    throw(ErrorException("Unidentifiable nucleotide"))
end

function to_rna(dna)
    map(choose,dna)
end

